app/Http/Controllers/CharacterController.php

```diff
@@ -196,9 +196,17 @@ public function show($guildSlug, $nameSlug)
         $guild         = request()->get('guild');
         $currentMember = request()->get('currentMember');
 
+        $guild->load('raids');

```

Comment:
Correct.

When we query the database with a specific model, we can load relationships for that model. In most cases, I eager load the relationships before working with them (as opposed to lazy loading). We have two primary ways of eager loading the relationships:

- Eager load when you first load the model: [with()](https://laravel.com/docs/7.x/eloquent-relationships#constraining-eager-loads)
- Eager lazy load at a later point: [load()](https://laravel.com/docs/7.x/eloquent-relationships#lazy-eager-loading) (this is still eager loading as we load it before we use it, but it's slightly lazy as we can do it at a later point; generally once we've determined whether or not we'll actually need it)

In this case, I've already loaded the guild object/model for the page we're viewing in order to perform some checks earlier on. I stored that guild in the request object for later access [here](https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Middleware/CheckGuildPermissions.php#L95). When I did that, I wasn't sure what page we'd be heading to, or what it would do with the guild object, so I didn't eager load the `raids` relationship or any other relationship just yet.

We end up in this controller, know we're going to need the `raids` relationship, and eager load it for later use.